src/main/java/uk/gov/ida/eidas/trustanchor/CountryTrustAnchor.java

```diff
@@ -26,21 +30,33 @@
 import com.nimbusds.jose.util.Base64;
 
 public class CountryTrustAnchor {
-  public static JWK make(X509Certificate certificate, String keyId) throws CertificateEncodingException {
-    RSAPublicKey publicKey = (RSAPublicKey) certificate.getPublicKey();
-    if (publicKey == null) {
+  public static JWK make(List<X509Certificate> certificates, String keyId) throws CertificateEncodingException {
+    List<PublicKey> invalidPublicKeys = certificates.stream()
+            .map(c -> c.getPublicKey())
+            .filter(key -> !(key instanceof RSAPublicKey))
+            .collect(Collectors.toList());
+
+    if (!invalidPublicKeys.isEmpty()) {
       throw new RuntimeException(String.format(
-        "Certificate public key in wrong format, got %s, expecting %s",
-        certificate.getPublicKey().getClass().getName(),
+        "Certificate public key(s) in wrong format, got %s, expecting %s",
+        String.join(" ", invalidPublicKeys.stream().map(key -> key.getClass().getName()).collect(Collectors.toList())),
         RSAPublicKey.class.getName()));
     }
 
-    Base64 base64cert = Base64.encode(certificate.getEncoded());
+    RSAPublicKey publicKey = (RSAPublicKey) certificates.get(0).getPublicKey();
```

Comment:
Have we specified or checked anywhere that the certificate chain is supplied in the right order?